src/pkg/exporter/exporter.go

```diff
@@ -0,0 +1,97 @@
+package exporter
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+
+	"github.com/goharbor/harbor/src/lib/log"
+)
+
+// Opt is the config of Harbor exporter
+type Opt struct {
```

Comment:
a few attributes are not used.

the attribute `ExporterMetricsEnabled` seems redundant.  We just skip starting the exporter if we don't want to enable it.

Reply:
`ExporterMetricsEnabled` is planned to use as a flag to decide if expose the metrics of exporter itself. Harbor info and exporter metrics are combined to expose at the moment. I can remove it now, and add it back in the future